### ISSUE TITLE
-i2jpackage (incorrectly) mangles CORBA::-types

### ISSUE THREAD

**Issue by magnemahre** [JacORB/JacORB#218]
I have an interface with a method which use a type from the CORBA:: namespace (included from orb.idl).

When I apply -i2jpackage to prefix my own defined interfaces, the CORBA::-types are affected.

module2.idl:
```
#include <orb.idl>

module A {
  interface myBaseInterface {
      CORBA::StringSeq f1();
  };
}; // module A

```
Command line:
`java  -cp /x/x/JacORB/idl-compiler/target/jacorb-idl-compiler-3.10-SNAPSHOT.jar org.jacorb.idl.parser -I/x/x/JacORB/idl/omg -i2jpackage :org.hytra module2.idl`

```
$ grep StringSeqHelper org/hytra/A/*
org/hytra/A/myBaseInterfacePOA.java:                            org.omg.org.hytra.CORBA.StringSeqHelper.write(_out,f1());
org/hytra/A/_myBaseInterfaceStub.java:                                  java.lang.String[] _result = org.omg.org.hytra.CORBA.StringSeqHelper.read(_is);

```

As you see, the class which should have been named org.omg.CORBA.StringSeqHelper has been renamed to org.omg.org.hytra.CORBA.StringSeqHelper, which I believe is a bug.

--Magne

**Comment by magnemahre** [comment]
The idlj-maven-plugin use -i2jpackage like this if you use the <packagePrefix> declaration:

https://github.com/mojohaus/idlj-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/idlj/JacorbTranslator.java#L190

**Comment by rnc** [comment]
Agreed that is not desired. Does running it as `idl -d xxxx -forceOverwrite -I<.....> -i2jpackage "A:org.hytra.A" <idl.file>` help?

**Comment by magnemahre** [comment]
In the actual case, I use nested modules - much of the idea was to utilize namespaces in the C++ code (built upon ACE/TAO).  The namespace structure doesn't look like the common Java structure, so I wanted to map it to something else (but could live with just prefixing the modules with "org.hytra...." (domain name is just an example).
The real code looks something like this:

module1.idl:
```
module A {
module B {
module C {
  struct myStruct {
      string strValue;
  };

  typedef sequence<myStruct> sqMyStruct;

  interface myInterface {
      void callMe ();
  };

}; // module C
}; // module B
}; // module A
```
which gives the namespace A::B::C structure in C++.
In Java, and with JacORB, I would like to either replace that or prefix it with the domain name.

When I compile this with  idl -i2jpackage A.B.C:org.hytra.C  module1.idl,
the sqMyStruct [Helper|Holder] is placed in "package org.hytra.C" (as I
expected), while the rest of the generated files are put in "package A.B.C"
(as was NOT expected)

I could also live with just prefixing (giving  `package org.hytra.A.B.C`), which why I tried the approach mentioned in the original bug report.  Having `package A.B.C` (without the domain name) will typically not fly due to coding standards here..

**Comment by rnc** [comment]
Agreed. Unfortunately this does indeed look like a bug with no available workaround.
